src/logic/calculate.js

```diff
@@ -0,0 +1,40 @@
+import operate from './operate';
+
+const calculate = (data, buttonName) => {
+  let { total, next, operation } = { ...data };
+
+  switch (buttonName) {
+    case 'AC':
+      total = '';
+      next = '';
+      operation = null;
+      break;
+    case '+/-':
+      if (next) {
+        next *= -1;
+      } else {
+        total *= -1;
+      }
+      break;
+    case '%':
+      if (next) {
+        next = operate(next, 100, '÷');
+      } else {
+        total = operate(total, 100, '÷');
+      }
+      break;
+    case '=':
+      total = operate(total, next, operation);
+      next = '';
+      operation = null;
+      break;
```

Comment:
Let's include the rest of the condition(buttonName). e.g. +, -, x, ÷.

![image](https://user-images.githubusercontent.com/31897434/113973695-a2ae8600-9834-11eb-909a-2f7246e9cce5.png)